tensorly/decomposition/_nn_cp.py

```diff
@@ -410,6 +552,139 @@ def fit_transform(self, tensor):
                                                  random_state=self.random_state,
                                                  verbose=self.verbose,
                                                  return_errors=True)
+
+
+        self.decomposition_ = cp_tensor
+        self.errors_ = errors
+        return self.decomposition_
+
+    def __repr__(self):
+        return f'Rank-{self.rank} Non-Negative CP decomposition.'
+
+
+
+class CP_NN_HALS(DecompositionMixin):
+    """Non-Negative Candecomp-Parafac decomposition via Alternating-Least Square
+
+        Computes a rank-`rank` decomposition of `tensor` [1]_ such that,
```

Comment:
This should be de-indented to match the beginning of the previous line.
Then either use:
```rst 
Computes a rank-`rank` decomposition of `tensor` [1]_ such that:: 

      ``tensor = [|weights; factors[0], ..., factors[-1] |]``.
``` 
if you want an explicit bloc quote or simply
```rst
Computes a rank-`rank` decomposition of `tensor` [1]_ such that ``tensor = [|weights; factors[0], ..., factors[-1] |]``.
```

I've been trying to harmonize the docstrings and iron these quirks out so it all renders nicely in the html/pdf documentation.